packages/auth/src/mfa/assertions/totp.ts

```diff
@@ -154,11 +157,21 @@ export class TotpMultiFactorAssertionImpl
   }
 
   /** @internal */
-  _finalizeSignIn(
-    _auth: AuthInternal,
-    _mfaPendingCredential: string
+  async _finalizeSignIn(
+    auth: AuthInternal,
+    mfaPendingCredential: string
   ): Promise<FinalizeMfaResponse> {
-    throw new Error('method not implemented');
+    _assert(
+      this.enrollmentId !== undefined && this.otp !== undefined,
```

Comment:
remove the check for `this.otp` since it's not marked optional on the class

Reply:
I had to add this as otherwise a different error was being thrown instead of auth/argument error when otp was passed as undefined. Also, it was stressed upon to add this as a check.

Reply:
Okay, we can leave this for now but let's discuss this at the next weekly. It was a conscious design decision in v9 that we rely as much as possible on the TypeScript type checker for this sort of stuff and avoid checking things that would already be an error in TypeScript compilation (it's not an optional parameter--if the developer doesn't pass it in it's on them).